upstream-sha256: d8f9a5da7b6cb452e25284ddfd7349c00c537f1f7faa40f6168bb93c91a95ada
actual-sha256: 390770fdbc903cea00b9c87f41ea9fd1175ae751502c70f24d84b81e145623fd
--- upstream.expected
+++ tolk.actual
@@ -6,72 +6,76 @@
     float4 val1 = (*((float4*)((data1_64+alu0))));
     float alu1 = -val0[0];
     float alu2 = ((alu1<-7.0f)?-7.0f:alu1);
-    float alu3 = (alu2*2.4554669857025146f);
-    _Bool alu4 = (alu3!=alu3);
-    float alu5 = ((alu3!=((float)(-__builtin_inff())))?alu3:0.0f);
-    float alu6 = (alu4?0.0f:alu5);
-    float alu7 = ((alu3!=((float)(__builtin_inff())))?alu6:0.0f);
-    float alu8 = ((alu7<0.0f)?-0.5f:0.5f);
-    int cast0 = ((int)((alu7+alu8)));
-    float alu9 = -val0[2];
-    float alu10 = ((alu9<-7.0f)?-7.0f:alu9);
-    float alu11 = (alu10*2.4554669857025146f);
-    _Bool alu12 = (alu11!=alu11);
-    float alu13 = ((alu11!=((float)(-__builtin_inff())))?alu11:0.0f);
-    float alu14 = (alu12?0.0f:alu13);
-    float alu15 = ((alu11!=((float)(__builtin_inff())))?alu14:0.0f);
-    float alu16 = ((alu15<0.0f)?-0.5f:0.5f);
-    int cast1 = ((int)((alu15+alu16)));
-    float alu17 = -val1[0];
-    float alu18 = ((alu17<-7.0f)?-7.0f:alu17);
-    float alu19 = (alu18*2.4554669857025146f);
-    _Bool alu20 = (alu19!=alu19);
-    float alu21 = ((alu19!=((float)(-__builtin_inff())))?alu19:0.0f);
-    float alu22 = (alu20?0.0f:alu21);
-    float alu23 = ((alu19!=((float)(__builtin_inff())))?alu22:0.0f);
-    float alu24 = ((alu23<0.0f)?-0.5f:0.5f);
-    int cast2 = ((int)((alu23+alu24)));
-    float alu25 = -val1[2];
-    float alu26 = ((alu25<-7.0f)?-7.0f:alu25);
-    float alu27 = (alu26*2.4554669857025146f);
-    _Bool alu28 = (alu27!=alu27);
-    float alu29 = ((alu27!=((float)(-__builtin_inff())))?alu27:0.0f);
-    float alu30 = (alu28?0.0f:alu29);
-    float alu31 = ((alu27!=((float)(__builtin_inff())))?alu30:0.0f);
-    float alu32 = ((alu31<0.0f)?-0.5f:0.5f);
-    int cast3 = ((int)((alu31+alu32)));
-    int alu33 = (cast0>>1);
-    int alu34 = (cast1>>1);
-    int alu35 = (cast2>>1);
-    int alu36 = (cast3>>1);
-    float alu37 = ((val0[1]<-7.0f)?-7.0f:val0[1]);
-    float alu38 = -alu37;
-    float alu39 = (alu7-((float)(cast0)));
-    float alu40 = (((alu3<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu39)+0.0013392627006396651f)*alu39)+0.009618384763598442f)*alu39)+0.055503472685813904f)*alu39)+0.24022644758224487f)*alu39)+0.6931471824645996f)*alu39)+1.0f)*__builtin_bit_cast(float, (int)(((alu33+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu33)+127)<<23)))));
-    float alu41 = ((alu3<-150.0f)?0.0f:alu40);
-    float alu42 = (alu4?((float)(__builtin_nanf(""))):alu41);
-    float alu43 = ((alu38<-7.0f)?-7.0f:alu38);
-    float alu44 = ((val0[3]<-7.0f)?-7.0f:val0[3]);
-    float alu45 = -alu44;
-    float alu46 = (alu15-((float)(cast1)));
-    float alu47 = (((alu11<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu46)+0.0013392627006396651f)*alu46)+0.009618384763598442f)*alu46)+0.055503472685813904f)*alu46)+0.24022644758224487f)*alu46)+0.6931471824645996f)*alu46)+1.0f)*__builtin_bit_cast(float, (int)(((alu34+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu34)+127)<<23)))));
-    float alu48 = ((alu11<-150.0f)?0.0f:alu47);
-    float alu49 = (alu12?((float)(__builtin_nanf(""))):alu48);
-    float alu50 = ((alu45<-7.0f)?-7.0f:alu45);
-    float alu51 = ((val1[1]<-7.0f)?-7.0f:val1[1]);
-    float alu52 = -alu51;
-    float alu53 = (alu23-((float)(cast2)));
-    float alu54 = (((alu19<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu53)+0.0013392627006396651f)*alu53)+0.009618384763598442f)*alu53)+0.055503472685813904f)*alu53)+0.24022644758224487f)*alu53)+0.6931471824645996f)*alu53)+1.0f)*__builtin_bit_cast(float, (int)(((alu35+127)<<23)))*__builtin_bit_cast(float, (int)((((cast2-alu35)+127)<<23)))));
-    float alu55 = ((alu19<-150.0f)?0.0f:alu54);
-    float alu56 = (alu20?((float)(__builtin_nanf(""))):alu55);
-    float alu57 = ((alu52<-7.0f)?-7.0f:alu52);
-    float alu58 = ((val1[3]<-7.0f)?-7.0f:val1[3]);
-    float alu59 = -alu58;
-    float alu60 = (alu31-((float)(cast3)));
-    float alu61 = (((alu27<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu60)+0.0013392627006396651f)*alu60)+0.009618384763598442f)*alu60)+0.055503472685813904f)*alu60)+0.24022644758224487f)*alu60)+0.6931471824645996f)*alu60)+1.0f)*__builtin_bit_cast(float, (int)(((alu36+127)<<23)))*__builtin_bit_cast(float, (int)((((cast3-alu36)+127)<<23)))));
-    float alu62 = ((alu27<-150.0f)?0.0f:alu61);
-    float alu63 = (alu28?((float)(__builtin_nanf(""))):alu62);
-    float alu64 = ((alu59<-7.0f)?-7.0f:alu59);
-    *((float4*)((data0_32+(Lidx0<<2)))) = (float4){-((alu18/(alu56+1.0f))*(1.0f-alu57)),-((alu26/(alu63+1.0f))*(1.0f-alu64)),-((alu2/(alu42+1.0f))*(1.0f-alu43)),-((alu10/(alu49+1.0f))*(1.0f-alu50))};
+    float alu3 = -alu2;
+    float alu4 = (alu3*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu5 = (alu4!=alu4);
+    float alu6 = ((alu4!=((float)(-__builtin_inff())))?alu4:0.0f);
+    float alu7 = (alu5?0.0f:alu6);
+    float alu8 = ((alu4!=((float)(__builtin_inff())))?alu7:0.0f);
+    float alu9 = ((alu8<0.0f)?-0.5f:0.5f);
+    int cast0 = ((int)((alu8+alu9)));
+    float alu10 = -val0[2];
+    float alu11 = ((alu10<-7.0f)?-7.0f:alu10);
+    float alu12 = -alu11;
+    float alu13 = (alu12*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu14 = (alu13!=alu13);
+    float alu15 = ((alu13!=((float)(-__builtin_inff())))?alu13:0.0f);
+    float alu16 = (alu14?0.0f:alu15);
+    float alu17 = ((alu13!=((float)(__builtin_inff())))?alu16:0.0f);
+    float alu18 = ((alu17<0.0f)?-0.5f:0.5f);
+    int cast1 = ((int)((alu17+alu18)));
+    float alu19 = -val1[0];
+    float alu20 = ((alu19<-7.0f)?-7.0f:alu19);
+    float alu21 = -alu20;
+    float alu22 = (alu21*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu23 = (alu22!=alu22);
+    float alu24 = ((alu22!=((float)(-__builtin_inff())))?alu22:0.0f);
+    float alu25 = (alu23?0.0f:alu24);
+    float alu26 = ((alu22!=((float)(__builtin_inff())))?alu25:0.0f);
+    float alu27 = ((alu26<0.0f)?-0.5f:0.5f);
+    int cast2 = ((int)((alu26+alu27)));
+    float alu28 = -val1[2];
+    float alu29 = ((alu28<-7.0f)?-7.0f:alu28);
+    float alu30 = -alu29;
+    float alu31 = (alu30*1.7020000219345093f*-1.4426950216293335f);
+    _Bool alu32 = (alu31!=alu31);
+    float alu33 = ((alu31!=((float)(-__builtin_inff())))?alu31:0.0f);
+    float alu34 = (alu32?0.0f:alu33);
+    float alu35 = ((alu31!=((float)(__builtin_inff())))?alu34:0.0f);
+    float alu36 = ((alu35<0.0f)?-0.5f:0.5f);
+    int cast3 = ((int)((alu35+alu36)));
+    int alu37 = (cast0>>1);
+    int alu38 = (cast1>>1);
+    int alu39 = (cast2>>1);
+    int alu40 = (cast3>>1);
+    float alu41 = ((val0[1]<-7.0f)?-7.0f:val0[1]);
+    float alu42 = -alu41;
+    float alu43 = ((val0[3]<-7.0f)?-7.0f:val0[3]);
+    float alu44 = -alu43;
+    float alu45 = ((val1[1]<-7.0f)?-7.0f:val1[1]);
+    float alu46 = -alu45;
+    float alu47 = ((val1[3]<-7.0f)?-7.0f:val1[3]);
+    float alu48 = -alu47;
+    float alu49 = (alu8-((float)(cast0)));
+    float alu50 = (alu17-((float)(cast1)));
+    float alu51 = (alu26-((float)(cast2)));
+    float alu52 = (alu35-((float)(cast3)));
+    float alu53 = (((alu4<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu49)+0.0013392627006396651f)*alu49)+0.009618384763598442f)*alu49)+0.055503472685813904f)*alu49)+0.24022644758224487f)*alu49)+0.6931471824645996f)*alu49)+1.0f)*__builtin_bit_cast(float, (int)(((alu37+127)<<23)))*__builtin_bit_cast(float, (int)((((cast0-alu37)+127)<<23)))));
+    float alu54 = ((alu4<-150.0f)?0.0f:alu53);
+    float alu55 = (alu5?((float)(__builtin_nanf(""))):alu54);
+    float alu56 = (((alu13<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu50)+0.0013392627006396651f)*alu50)+0.009618384763598442f)*alu50)+0.055503472685813904f)*alu50)+0.24022644758224487f)*alu50)+0.6931471824645996f)*alu50)+1.0f)*__builtin_bit_cast(float, (int)(((alu38+127)<<23)))*__builtin_bit_cast(float, (int)((((cast1-alu38)+127)<<23)))));
+    float alu57 = ((alu13<-150.0f)?0.0f:alu56);
+    float alu58 = (alu14?((float)(__builtin_nanf(""))):alu57);
+    float alu59 = (((alu22<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu51)+0.0013392627006396651f)*alu51)+0.009618384763598442f)*alu51)+0.055503472685813904f)*alu51)+0.24022644758224487f)*alu51)+0.6931471824645996f)*alu51)+1.0f)*__builtin_bit_cast(float, (int)(((alu39+127)<<23)))*__builtin_bit_cast(float, (int)((((cast2-alu39)+127)<<23)))));
+    float alu60 = ((alu22<-150.0f)?0.0f:alu59);
+    float alu61 = (alu23?((float)(__builtin_nanf(""))):alu60);
+    float alu62 = (((alu31<128.0f)!=1)?((float)(__builtin_inff())):(((((((((((((0.00015359208919107914f*alu52)+0.0013392627006396651f)*alu52)+0.009618384763598442f)*alu52)+0.055503472685813904f)*alu52)+0.24022644758224487f)*alu52)+0.6931471824645996f)*alu52)+1.0f)*__builtin_bit_cast(float, (int)(((alu40+127)<<23)))*__builtin_bit_cast(float, (int)((((cast3-alu40)+127)<<23)))));
+    float alu63 = ((alu31<-150.0f)?0.0f:alu62);
+    float alu64 = (alu32?((float)(__builtin_nanf(""))):alu63);
+    float alu65 = ((alu42<-7.0f)?-7.0f:alu42);
+    float alu66 = ((alu44<-7.0f)?-7.0f:alu44);
+    float alu67 = ((alu46<-7.0f)?-7.0f:alu46);
+    float alu68 = ((alu48<-7.0f)?-7.0f:alu48);
+    *((float4*)((data0_32+(Lidx0<<2)))) = (float4){((alu21/(alu61+1.0f))*(1.0f-alu67)),((alu30/(alu64+1.0f))*(1.0f-alu68)),((alu3/(alu55+1.0f))*(1.0f-alu65)),((alu12/(alu58+1.0f))*(1.0f-alu66))};
   }
 }
